tests/functional/collection/test_collection_list.py

```diff
@@ -30,6 +30,16 @@ def test_collection_list(run_line, add_gcs_login, base_command):
         assert name in result.stdout
 
 
+def test_collection_list_limit(run_line, add_gcs_login, base_command):
```

Comment:
This tests the limit, but is there a test that exercises the pagination across a larger limit than what's returned?

For example, if the first call returns 16 but the requested limit is 25, is there a test that exercises the pagination and guarantees that the next call, also returning 16, is clipped to only 9 entries?

Reply:
There isn't, at least not directly. I actually didn't see tests exercising similar scenario elsewhere this paradigm is used in the CLI, so hadn't included it here and instead opted to manually exercise it. I'll look into it.

Reply:
Added!